exporter/signalfxexporter/factory.go

```diff
@@ -64,13 +66,34 @@ func createDefaultConfig() configmodels.Exporter {
 		AccessTokenPassthroughConfig: splunk.AccessTokenPassthroughConfig{
 			AccessTokenPassthrough: true,
 		},
-		SendCompatibleMetrics: false,
-		TranslationRules:      nil,
-		DeltaTranslationTTL:   3600,
-		Correlation:           correlation.DefaultConfig(),
+		DeltaTranslationTTL: 3600,
+		Correlation:         correlation.DefaultConfig(),
 	}
 }
 
+func customUnmarshaler(componentViperSection *viper.Viper, intoCfg interface{}) (err error) {
+	if componentViperSection == nil {
+		// Nothing to do if there is no config given.
+		return nil
+	}
+
+	if err = componentViperSection.Unmarshal(intoCfg); err != nil {
+		return err
+	}
+
+	config := intoCfg.(*Config)
+
+	// custom unmarhalling is required to get []translation.Rule, the default
+	// unmarshaller only supports string slices. If translations_config is not
```

Comment:
I don't quite understand the comment. In the case where the user has specified translation rules how are those getting unmarshalled then? I don't see any custom code for doing the unmarshaling of them. Are we not able to configure the default translation rules in the default config created by the factory?

Reply:
The initial impression I got was the default translations aren't being picked up correctly, but I was mistaken. The issue I ran into was, the default unmarshaler not being able to override default `translation_rules`. So, instead of setting it in the factory, I set it in the customer unmarshaler if it is unset.

Reply:
Updated the comment.